libmono-native-compat.dylib missing from macOS packaging

Library is missing from the `mkbundle` redistribution (like on Linux), and need to remove the `$mono_libdir/`s from the `config`. This will need to be fixed in https://github.com/OpenRA/OpenRALauncherOSX/

Regression from #17170.

```
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'OpenRA.Platform' threw an exception. ---> System.DllNotFoundException: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/../lib/libmono-native-compat.dylib assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Interop+Sys.GetNonCryptographicallySecureRandomBytes(byte*,int)
  at Interop.GetRandomBytes (System.Byte* buffer, System.Int32 length) [0x00000] in <b814b509d4ad406fb40c6c93e38929e7>:0 
  at System.Guid.NewGuid () [0x00000] in <b814b509d4ad406fb40c6c93e38929e7>:0 
  at OpenRA.Platform..cctor () [0x00000] in <ed3af7fbe29b438f9d5e479d4a9cd718>:0 
   --- End of inner exception stack trace ---
  at OpenRA.Log+<FilenamesForChannel>d__1.MoveNext () [0x0001a] in <ed3af7fbe29b438f9d5e479d4a9cd718>:0 
  at OpenRA.Log.AddChannel (System.String channelName, System.String baseFilename, System.Boolean isTimestamped) [0x000a0] in <ed3af7fbe29b438f9d5e479d4a9cd718>:0 
  at OpenRA.ExceptionHandler.HandleFatalError (System.Exception ex) [0x00027] in <ed3af7fbe29b438f9d5e479d4a9cd718>:0 
  at OpenRA.Program+<>c.<Main>b__0_0 (System.Object _, System.UnhandledExceptionEventArgs e) [0x00006] in <ed3af7fbe29b438f9d5e479d4a9cd718>:0 
2019-10-07 20:39:16.573 OpenRA[2476:221126] launchgame exited with code 1
```

--- a/packaging/osx/buildpackage.sh
+++ b/packaging/osx/buildpackage.sh
@@ -9,7 +9,7 @@ if [[ "$OSTYPE" != "darwin"* ]]; then
 	command -v genisoimage >/dev/null 2>&1 || { echo >&2 "macOS packaging requires genisoimage."; exit 1; }
 fi
 
-LAUNCHER_TAG="osx-launcher-20191003"
+LAUNCHER_TAG="osx-launcher-20191007"
 
 if [ $# -ne "2" ]; then
 	echo "Usage: $(basename "$0") tag outputdir"

--- a/thirdparty/fetch-thirdparty-deps-osx.sh
+++ b/thirdparty/fetch-thirdparty-deps-osx.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-LAUNCHER_TAG="osx-launcher-20191003"
+LAUNCHER_TAG="osx-launcher-20191007"
 
 download_dir="${0%/*}/download/osx"
 mkdir -p "$download_dir"

